src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.cs

```diff
@@ -225,6 +226,8 @@ private void ValidateProcessedBlock(Block suggestedBlock, ProcessingOptions opti
         _receiptsTracer.SetOtherTracer(blockTracer);
         _receiptsTracer.StartNewBlockTrace(block);
 
+        BeaconBlockRootPrecompile.Instance.SetupBeaconBlockRootPrecompileState(spec.IsBeaconParentBlockRootAvailable, _stateProvider, block.ParentBeaconBlockRoot, block.Timestamp);
```

Comment:
Can we hide it behind something more generic and spec-based similarly to `_withdrawalProcessor.ProcessWithdrawals(block, spec);` And then don't pass spec parameters to the precompile.